k4GaudiPandora/include/CalorimeterHitType.h

```diff
@@ -0,0 +1,168 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef CalorimeterHitType_h
+#define CalorimeterHitType_h 1
+
+#include <iostream>
+
+/** Helper class for decoding/encoding lcio::CalorimeterHit types for the ILD
+ *  detector. The encoding is: caloType + 10 * caloID + 1000 * layout + 10000 * layerNum <br>
+ *  (see enums: CaloType, CaloID and Layout for possible values).<br>
+ *  Example usage: <br>
+ *  <pre>
+ *     lcio::CalorimeterHit* cHit = .... ;
+ *
+ *     // set the type (e.g. in digitization )
+ *     cHit->setType( CHT( CHT::em ,  CHT::ecal , CHT::plug , 12 ) ) ;
+ *
+ *     ...
+ *
+ *     CHT cht = cHit->getType() ;
+ *
+ *     //   sum energies for electromagentic, hadronic and tailcatcher:
+ *     if( cht.is( CHT::em ) )
+ *          e_em +=  cHit->getEnergy() ;
+ *     else
+ *       if ( cht.is(CHT::had ) )
+ *          e_had += cHit->getEnergy() ;
+ *       else
+ *          e_muon += cHit->getEnergy() ;
+ *
+ *     // use only EcalPlug hits:
+ *     if( cht.is( CHT::ecal) && cht.is( CHT::plug) )
+ *
+ *     // get the layer number (e.g. for calibration or clustering)
+ *     unsigned l = cht.layer() ;
+ *     // or directly :
+ *     unsigned l = CHT(  cHit->getType() ).layer()  ;
+ *
+ *     // detailed print:
+ *     std::cout <<  CHT(  cHit->getType() ) << std::endl ;
+ *
+ *  </pre>
+ *
+ *  F.Gaede, DESY, 12/2008
+ */
+
+class CHT {
+public:
+  /** calorimeter types */
+  typedef enum { em = 0, had = 1, muon = 2 } CaloType;
+
+  /** calo ids - specific to ILD */
+  typedef enum { unknown = 0, ecal = 1, hcal = 2, yoke = 3, lcal = 4, lhcal = 5, bcal = 6 } CaloID;
+
+  /** calo layout / subdetector  */
+  typedef enum { any = 0, barrel = 1, endcap = 2, plug = 3, ring = 4 } Layout;
+
+  /** C'tor for initialization from CalorimeterHit::getType()  */
+  CHT(int type) : _type(type) {}
+
+  /** C'tor  for encoding the calo type inforamtion  */
+  CHT(CaloType c, CaloID n, Layout l, unsigned lay) : _type(c * fCaloType + n * fCaloID + l * fLayout + lay * fLayer) {}
+
+  /** calorimeter type: CHT::em , CHT::had, CHT::muon */
+  CaloType caloType() const { return (CaloType)(_type % fCaloID); }
+
+  /** calo ID - see enum CaloID for allowed values */
+  CaloID caloID() const { return (CaloID)((_type % fLayout) / fCaloID); }
+
+  /** calo layout - see enum layout for allowed values */
+  Layout layout() const { return (Layout)((_type % fLayer) / fLayout); }
+
+  /** calo layer of hit  */
+  unsigned layer() const { return unsigned(_type) / fLayer; }
+
+  bool is(CaloType t) const { return caloType() == t; }
+
+  bool is(CaloID n) const { return caloID() == n; }
+
+  bool is(Layout l) const { return layout() == l; }
+
+  /** automatic conversion to int */
+  operator int() const { return _type; }
+
+  /** explicit conversion to int */
+  int toInt() const { return _type; }
+
+protected:
+  int _type;
```

Comment:
```suggestion
  int m_type;
```
Member variables starting with an `_` are a slippery slope as at least part of them are reserved for compiler and standard library internal names.